modules/generic_worker/templates/run-generic-worker.sh.erb

```diff
@@ -0,0 +1,69 @@
+
+# If this file exists when the worker starts up, and its content is equal
+# to numberOfTasksToRun, the worker won't exit and machine won't reboot.
+rm -f tasks-resolved-count.txt
+
+export PATH=/bin:/sbin:/usr/bin:/usr/sbin:/usr/local/bin
+source /usr/local/share/generic-worker/bugzilla-utils.sh
+
+# First run the generic-worker, passing through any arguments handed to this
+# wrapper script...
+/usr/local/bin/generic-worker "$@" 2>&1 | logger -t generic-worker -s
+
+exitstatus="${PIPESTATUS[0]}"
+
+export TASKCLUSTER_CLIENT_ID="<%= @quarantine_client_id %>"
+export TASKCLUSTER_ACCESS_TOKEN="<%= @quarantine_access_token %>"
+
+if [ $exitstatus -eq 69 ]; then
+# If generic-worker considers that the host environment is in an invalid/corrupted
+# state (for example, it cannot write to disk), it will return the exit code 69.
+#
+# More information about generic-worker exit codes can be seen here:
+# https://docs.taskcluster.net/docs/reference/workers/generic-worker#set-up-your-env
+#
+# If this happens, we make a call to Taskcluster Queue to quarantine the worker, so
+# that no further tasks will be provided to the worker until the quarantine expires.
+# This prevents the worker from burning through tasks that it is not able to execute
+# properly.
+    if [ -f <%= @user_homedir %>/quarantined ]; then
+        echo "Worker is quarantined"
+    else
+        echo "Quarantine worker"
+        # quarantine the worker using github.com/mozilla-platform-ops/quarantine-worker CLI
+        /usr/local/bin/quarantine-worker releng-hardware "<%= @worker_type %>" "<%= @worker_group %>" "<%= @hostname %>" 720h5s
+
+        # create a semaphore file
+        echo "quarantined worker" > <%= @user_homedir %>/quarantined
+
+        # Submit a new bug to Bugzilla that reports that the worker has been quarantined,
+        # in order to alert CI Duty to the failure, track status and facilitate resolution.
+        create_bug_self_failure
+    fi
+elif [ $exitstatus -gt 69 -o $exitstatus -lt 69 ]; then
+    # If exitstatus is different by 69 (grater than 69 or lower than 69)
+    # The worker was removed from the quarantin and run with success a job
+    # Remove quarantined file, if file exist
+    if [ -f <%= @user_homedir %>/quarantined ]; then
+        /bin/rm -rf <%= @user_homedir %>/quarantined
+    fi
+    # Self resolve the bug and remove bug_id file if file exist
+    if [ -f <%= @user_homedir %>/bug_id ]; then
+        resolve_bug_self_failure
+        /bin/rm -rf <%= @user_homedir %>/bug_id
+    fi
+    # Now reboot the machine immediately (time costs money)
+    <%= @reboot_command %>
+    # Sleep to prevent this script from terminating naturally, and launchd restarting
+    # it. Instead, the shutdown should cause this script to terminate (so it won't
+    # really sleep for 2 mins). If shutdown doesn't kick in within 2 mins, it is sane
+    # for this script to exit, and allow launchd to fire up the worker again.
+    /bin/sleep 120
+fi
```

Comment:
This file was modified. See https://github.com/mozilla-releng/build-puppet/pull/409/files